### DISCUSSION TITLE
How to use the Tauri JS API to copy a directory to another location?

### DISCUSSION THREAD

**Discussion by bling-yshs** [tauri-apps/tauri#8198] [Q&A]
I tried using `import {copyFile} from "@tauri-apps/api/fs";`, but it's apparent that this method can only copy files. 

Is it only possible to achieve this through rust then?

**Comment by FabianLars** [comment]
If we only talk about Rust std (so no external crates) there's also no explicit method for this. The general idea is to recursively use readDir. Check the type of the entry and copy files directly or run readDir again if it's a folder. Basically like this: https://stackoverflow.com/a/65192210 (This is for rust but it's the same in JS).

**Reply by bling-yshs** [reply]
thank you!

**Comment by bling-yshs** [comment]
In the end, I solve it with rust.

```shell
cargo add fs_extra
```
```rust

#[tauri::command]
async fn copy_directory(source: String, destination: String) -> DataResponse<String> {
    let mut options = fs_extra::dir::CopyOptions::new();
    options.copy_inside = true;
    return match fs_extra::dir::copy(source, destination, &options) {
        Ok(_) => {
            DataResponse::fast_success()
        }
        Err(e) => {
            DataResponse::failure(e.to_string())
        }
    };
}
```
